crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -563,35 +563,37 @@ impl CheckpointExecutor {
 
         if let Some(checkpoint) = checkpoint {
             if checkpoint.epoch() == cur_epoch {
-                if let Some((change_epoch_execution_digests, change_epoch_tx)) =
-                    extract_end_of_epoch_tx(
-                        checkpoint,
-                        self.authority_store.clone(),
-                        self.checkpoint_store.clone(),
-                        epoch_store.clone(),
-                    )
-                {
-                    let change_epoch_tx_digest = change_epoch_execution_digests.transaction;
+                let end_of_epoch_txns = extract_end_of_epoch_txns(
+                    checkpoint,
+                    self.authority_store.clone(),
+                    self.checkpoint_store.clone(),
+                    epoch_store.clone(),
+                );
+                let end_of_epoch = !end_of_epoch_txns.is_empty();
+                for (digests, end_of_epoch_tx) in end_of_epoch_txns.into_iter() {
+                    let end_of_epoch_tx_digest = digests.transaction;
 
                     info!(
                         ended_epoch = cur_epoch,
                         last_checkpoint = checkpoint.sequence_number(),
-                        "Reached end of epoch, executing change_epoch transaction",
+                        "Reached end of epoch, executing end_of_epoch transaction",
```

Comment:
Could we add the digest to the log line here given that there are now more than one?